lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -354,9 +355,11 @@ static LogicalResult canonicalizePrimOp(
     resultValue = rewriter.create<PadPrimOp>(op->getLoc(), resultValue, width);
 
   // Insert a cast if this is a uint vs. sint or vice versa.
-  if (type.isa<SIntType>() && resultValue.getType().isa<UIntType>())
+  if (firrtl::type_isa<SIntType>(type) &&
+      firrtl::type_isa<UIntType>(resultValue.getType()))
     resultValue = rewriter.create<AsSIntPrimOp>(op->getLoc(), resultValue);
-  else if (type.isa<UIntType>() && resultValue.getType().isa<SIntType>())
+  else if (firrtl::type_isa<UIntType>(type) &&
+           resultValue.getType().isa<SIntType>())
```

Comment:
Shouldn't the second `isa` be `type_isa` too? For all lines containing multiple cast-checks, please vet they are handled consistently or add a comment indicating why they shouldn't be.

I wish it wasn't unreasonable to ask for test coverage of this large change, but I understand that would basically require duplicating (and more) all our existing test infra.  Please think about adding at least some tests for these tings, however, especially as they're very very easy to get wrong / miss even in review.  Downside of this approach is carrying that burden collectively.